### DISCUSSION TITLE
Is there any way to register a dumper for a cursor while fetching the data and after executing a query?

### DISCUSSION THREAD

**Discussion by khushboovashi** [psycopg/psycopg#881] [General]
Is there any way to register a dumper for a cursor while fetching the data and after executing a query?

**Comment by dvarrazzo** [comment]
It's in a grey area of the Transformer life cycle. Dumpers registered will obviously be used from the following query on, because the dumper transforms the query parameter. If you mean to register a loader, that's unspecified: it depends whether the loaders are instantiated upon receiving the result or upon parsing the first data tuple, and shouldn't be rely on.

Can you specify better what you would like to do?

**Comment by khushboovashi** [comment]
My scenario is, 
- Registered a loader (connection.adapters.register_loader...)
- Execute a query
- fetch data
- Requirement: Want to register another loader for the same async cursor
- fetch data from the same async cursor (without executing the query)

**Reply by dvarrazzo** [reply]
It might be possible to specify better this behaviour. At the moment is not specified and not tested. Adding a ticket to consider acting on it.

**Reply by khushboovashi** [reply]
Thanks.
